metadata_service/dao/dataset.py

```diff
@@ -88,7 +96,9 @@ async def update_dataset(dataset_id: str, data: Dict) -> Dict:
       The updated Dataset object
 
     """
-    collection = await get_collection(COLLECTION_NAME)
-    await collection.update_one({"id": dataset_id}, {"$set": data})
+    db_connect = DBConnect()
+    collection = await db_connect.get_collection(COLLECTION_NAME)
+    await collection.update_one({"id": dataset_id}, {"$set": data.dict()})  # type: ignore
     dataset = await get_dataset(dataset_id)
+    db_connect.close_db()
```

Comment:
No need to close the db connection. This is already being taken care of by the `get_dataset` method in the line above.

Reply:
moved the close method one line above so as to match the get and close connection